src/lumen/activity/context.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "lumen/activity/activity.hpp"
+
+#include <memory>
+
+namespace lumen::activity {
+
+/// The context class for displayable activities.
+class Context {
+public:
+    /** Context constructor.
+     *
+     * \param type The type of the activity to be set.
+     */
+    explicit Context(Type type = Type::none);
+
+    [[nodiscard]] Activity* get_activity() const;
```

Comment:
Does it make sense to have this be `const`? I feel like the argument for pass by reference return values could also apply to this.

Reply:
Fair enough.